middle.hpp

```diff
@@ -14,26 +14,42 @@
 
 struct options_t;
 
+/**
+ * @brief object which stores OSM node/ways/relations from the input file
```

Comment:
Enable JAVADOC_AUTOBRIEF in the configuration file and remove brief here. autobrief is better readable in the code. Also choose either @-style or /-style.